flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/WaitingForResources.java

```diff
@@ -90,13 +107,40 @@ public Logger getLogger() {
 
     @Override
     public void notifyNewResourcesAvailable() {
-        if (context.hasEnoughResources(desiredResources)) {
+        checkDesiredOrSufficientResourcesAvailable();
+    }
+
+    private void checkDesiredOrSufficientResourcesAvailable() {
+        if (context.hasDesiredResources(desiredResources)) {
             createExecutionGraphWithAvailableResources();
+            return;
         }
+
+        if (context.hasSufficientResources()) {
```

Comment:
One minor idea could be to unify `hasDesiredResources` and `hasSufficientResources`. I think both methods do effectively the same and could signal the different results with a more expressive return type.

Reply:
If its okay for you, for the sake of time, I would address this later.

Reply:
Alright. Sounds good.